### DISCUSSION TITLE
sync faied to index

### DISCUSSION THREAD

**Discussion by cyberb** [photoprism/photoprism#4322] [General]
Does anyone know what can cause sync to fail to bring new pictures from Nextcloud over webdav, it worked to bring the original set of files, but fails to bring new files

Photoprism version: 240523

```
level=info msg="sync: downloaded /Boris Phone/Camera/2024/06/IMG_20240606_070247.jpg from Nextcloud
level=info msg="sync: indexing /Boris Phone/Camera/2024/06/IMG_20240606_070247.jpg and related file
level=debug msg="index: updating counts"
level=debug msg="metadata: no gps tags in IMG_20240606_070247.jpg (exif gps-info)"
level=debug msg="media: exiftool json files disabled"
level=debug msg="counts: updated 219 places [20.81924ms]"
level=debug msg="counts: updated 16 subjects [47.015946ms]"
level=debug msg="counts: updated 197 labels [63.760804ms]"
level=debug msg="index: updating covers"
level=debug msg="covers: updated 0 albums [643.163µs]"
level=info msg="media: generated 10 thumbnails for 'Boris Phone/Camera/2024/06/IMG_20240606_070247.
level=info msg="media: IMG_20240606_070247.jpg was taken at 2024-06-06 07:02:51 +0000 UTC (meta)"
level=error msg="photo: database is locked (update date fields)"
level=error msg="photo: database is locked (update date fields)"
level=error msg="photo: database is locked (update date fields)"
level=debug msg="covers: updated 1,032 folders [21.339271084s]"
level=debug msg="covers: updated 151 months [2.111195231s]"
level=error msg="index: failed to create 'Boris Phone/Camera/2024/06/IMG_20240606_070247.jpg'"
level=error msg="index: failed to add main jpg file 'Boris Phone/Camera/2024/06/IMG_20240606_070247
level=debug msg="covers: updated 205 labels [9.98329287s]"
level=debug msg="covers: updated 8 subjects [11.643001ms]"
level=debug msg="index: updating metadata"

```

**Comment by graciousgrey** [comment]
It looks like there is a problem with your database. Are you using mariadb or sqlite?

https://docs.photoprism.app/getting-started/faq/#should-i-use-sqlite-mariadb-or-mysql
https://docs.photoprism.app/getting-started/troubleshooting/sqlite/#locking-errors

**Comment by cyberb** [comment]
Sqlite on SSD, is there any command to check the db file consistency?

**Comment by cyberb** [comment]
I can open db file and dump, seems to be not corrupted. It is 300mb I guess I better switch to mariadb? It would be good to know if that will actually fix this problem :)

**Reply by lastzero** [reply]
If it's caused by locking, then probably yes.

**Comment by cyberb** [comment]
Complete rescan fixes this issue (missing photos are visible now)
Let me check if new photos are coming from nextcloud now.

**Comment by cyberb** [comment]
> If it's caused by locking

Is there any way to know for sure it is happening?
Would debug log level help?

**Comment by cyberb** [comment]
Because for me it looks like nextcloud sync brings files that cannot be indexed incrementally if there such a thing. I would assume db locking issues to give more errors in the log and be more random.
But of course it would be easier just to migrate to mariadb and check again, will give it a go in couple days.
